### ISSUE TITLE
kapp-controller Pod stuck in CLBO and will never recover once the kapp-controller-sidecarexec container is SIGKILL’ed

### ISSUE THREAD

**Issue by scarlettqu** [carvel-dev/kapp-controller#878]
**What steps did you take:**
Over the uptime, noticed kapp-controller-sidecarexec is in CLBO due to /etc/kappctrl-mem-tmp/sidecarexec.sock: bind: address already in use.
I was able to reproduce the issue by sending a SIGKILL signal to the container using `kill -9 <container pid>`.

**What happened:**
The Pod stuck in CLBO and the package resource is not recognized.
```
root@422f0e147ceebf2c4b41fd0b86093bc2 [ ~ ]# kubectl get po -n vmware-system-appplatform-operator-system
NAME                                                    READY   STATUS             RESTARTS       AGE
kapp-controller-786f8cf789-pv5m7                        1/2     CrashLoopBackOff   3 (28s ago)    32m

root@422f0e147ceebf2c4b41fd0b86093bc2 [ ~ ]# kubectl logs -n vmware-system-appplatform-operator-system kapp-controller-786f8cf789-pv5m7 kapp-controller-sidecarexec
{"level":"info","ts":1663028728.5450194,"logger":"kc-sidecarexec","msg":"start sidecarexec","version":"0.38.4"}
{"level":"info","ts":1663028728.5464313,"logger":"kc-sidecarexec","msg":"starting zombie reaper"}
{"level":"error","ts":1663028728.5473244,"logger":"kc-sidecarexec","msg":"Serving RPC","error":"Listening RPC: listen unix /etc/kappctrl-mem-tmp/sidecarexec.sock: bind: address already in use","stacktrace":"runtime.main\n\truntime/proc.go:250"}

root@422f0e147ceebf2c4b41fd0b86093bc2 [ ~ ]# kubectl get packages -A
error: the server doesn't have a resource type "packages"
```

**What did you expect:**
I expect the kapp-controller Pod can auto recover without any manual steps.

**Anything else you would like to add:**
1. Initial analysis: the issue comes from emptyDir: kapp-controller-sidecarexec is using emptyDir Volume for the /etc/kappctrl-mem-tmp/sidecarexec.sock. Since emptyDir persists among container crashes, the socket will not be released after the crash. Then the socket cannot bind after kapp-controller-sidecarexec restarts, which cause the CLBO.
2. Various reasons can cause the container be SIGKILL'ed (OOMKilled error, delete without grace period, etc), this means any kapp-controller installation would hit this issue (sooner or later).
3. Manually restart the Pod can recover from the issue. But I still expect the Pod can recover without any manual stapes because the customers might not have the permissions for such solution on the cluster.

**Environment:**

- kapp Controller version (execute `kubectl get deployment -n kapp-controller kapp-controller -o yaml` and the annotation is `kbld.k14s.io/images`): v0.38.4
- Kubernetes version (use `kubectl version`): 1.23

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by scarlettqu** [comment]
Thanks for the followup. Remove the file can unblock from the issue, but I still expect the Pod to recover without any manual operations on the cluster since the customer may not have the permission for this kind of solutions on the cluster. 

For example, one possible solution may be to delete the socket file if it exists before starting sidecar-exec listener?

**Comment by cppforlife** [comment]
this is addressed in v0.41.2
